### ISSUE TITLE
Finetune with additional layers

### ISSUE THREAD

**Issue by Checkmate986212** [tensorflow/tensorflow#32732]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu16.04
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): 1.10
- Python version: 3.6
- CUDA/cuDNN version: 9.0
- GPU model and memory: 1070TI

I wonder how to finetune a model with several additional layers
I have a set of checkpoint, including .ckpt and .meta

I add two deconvolution layers to a previous model
```
flow_pred_x = tf.layers.conv2d_transpose(flow_x, filters=1, kernel_size=8, strides=4, padding='same', use_bias=False, name='slice_deconv_x')
flow_pred_y = tf.layers.conv2d_transpose(flow_y, filters=1, kernel_size=8, strides=4, padding='same', use_bias=False, name='slice_deconv_y')
```
<br>
I tried to use this at first<br>

'''
saver = tf.train.Saver()
saver.restore(session, './model.ckpt')
```
However, the error is unmatch<br>
Then I tried to save as .pb
```
constant_graph = graph_util.convert_variables_to_constants(self.sess, input_graph_def=self.sess.graph_def, output_node_names=['pwcnet/ctxt/refined_flow2', 'pwcnet/flow_x', 'pwcnet/flow_y'])
with tf.gfile.FastGFile('./modelflowxy.pb', mode='wb') as f:
       f.write(constant_graph.SerializeToString())
```
And load the .pb and add layers<br>
```
import tensorflow as tf
import numpy as np
from tensorflow.python.tools.inspect_checkpoint import print_tensors_in_checkpoint_file
with tf.gfile.FastGFile(r'./modelflowxy.pb', 'rb') as f:
    graph_def = tf.GraphDef()
    graph_def.ParseFromString(f.read())
    _ = tf.import_graph_def(graph_def, name='')
with tf.Session() as session:
   flow_x = tf.get_default_graph().get_tensor_by_name('pwcnet/flow_x:0')
    flow_y = tf.get_default_graph().get_tensor_by_name('pwcnet/flow_y:0')
    init = np.load('./slice_conv_x_0_w.npy')
    init = tf.constant_initializer(init)
    flow_pred_x = tf.layers.conv2d_transpose(flow_x, filters=1, kernel_size=8, strides=4, padding='same', use_bias=False, name='slice_deconv_x')
    flow_pred_y = tf.layers.conv2d_transpose(flow_y, filters=1, kernel_size=8, strides=4, padding='same', use_bias=False, name='slice_deconv_y')
    flow_pred = tf.concat([flow_pred_x, flow_pred_y], axis=3, name='flow_pred')
    init = tf.global_variables_initializer()
    session.run(init)
    saver = tf.train.Saver(var_list=tf.trainable_variables())
    saver.save(session, './ckpt_test2/all.ckpt')
```

It doesn't work too, and the layers in the previous model unmatch
And I check the new checkpoint, and find that only two newly added layers are recorded.<br>
So, I sincerely hope if anyone could tell me how to finetune model with additional layers

**Comment by ymodak** [comment]
See https://www.tensorflow.org/tutorials/images/transfer_learning for your use case.
